docs/architecture/adr-009-v6-ics27-msgserver.md

```diff
@@ -0,0 +1,110 @@
+
+The controller module will now claim the channel capability in `OnChanOpenInit`. 
+Underlying applications will be passed a nil capability in `OnChanOpenInit` 
+
+Channel capability migrations will be added in two steps:
+- Upgrade handler migration which modifies the channel capability owner from the underlying app to the controller module
+- ICS 27 module automatic migrations which assert the upgrade handler channel capability migration has been performed successfully
+
+See issue [#2033](https://github.com/cosmos/ibc-go/issues/2033)
+
+### Middleware enabled channels
+
+In order to maintain backwards compatibility and avoid requiring underlying application developers to account for interchain accounts they did not register, a boolean mapping has been added to track the behaviour of how an account was created. 
+
+If the account was created via the legacy API, then the underlying application callbacks will be executed
+
+If the account was created with the new API (message server), then the underlying application callbacks will not be executed.
+
+See issue [#2145](https://github.com/cosmos/ibc-go/issues/2145)
+
+### Future considerations
+
+ADR 008 proposes the creation of a middleware which enables callers of an IBC packet send to perform application logic in conjunction with the IBC application. 
+The underlying application can be removed at the availablity of such a middleware as that will be the preferred method for executing application logic upon a ICS 27 packet send.
+
+### Miscellanous 
+
+In order to avoid import cycles, the genesis types have been moved to their own directory.
+A new protobuf package has been created for the genesis types.
+
+See pr [#2133](https://github.com/cosmos/ibc-go/pull/2133)
+
+An additional field has been added to the `ActiveChannel` type to store the `IsMiddlewareEnabled` field upon genesis import/export.
+
+See issue [#2165](https://github.com/cosmos/ibc-go/issues/2165)
+
+## Consequences
+
+### Positive
+
+- default authentications module are provided (x/auth, x/group, x/gov) 
+- any SDK authentication module may now be used with ICS 27
+- separation of authentication from appplication logic in relation to ICS 27
+- minimized disruption to existing development around ICS 27 controller module
+- underlying applications no longer have to handle capabilities
+- removal of the underlying application upon the creation of ADR 008 may be done in a minimally disruptive fashion
+- only underlying applications which registered the interchain account will perform application logic for that account (underlying applications do not need to be cogniscent of accounts it did not register)
+
+### Negative
+
+- the security model has been reduced to that of the SDK. SDK modules may send packets for any interchain account.
+- additional maintence of the messages added and the middleware enabled flag
+- underlying applications which will become ADR 008 module are not required to be aware of accounts they did not register
```

Comment:
```suggestion
- underlying applications which will become ADR 008 modules are not required to be aware of accounts they did not register
```

Reply:
Actually, I think I don't understand this point...

Reply:
If you didn't have the is middleware enabled mapping, underlying applications would need to account for accounts they did not register. This is required behaviour for ADR 008 modules (which some underlying applications will become)